api/net/Areas/Editor/Controllers/StorageController.cs

```diff
@@ -49,216 +55,627 @@ public StorageController(IOptions<StorageOptions> options)
     /// <summary>
     /// Fetch an array of files and directories at the specified path.
     /// </summary>
+    /// <param name="locationId"></param>
     /// <param name="path"></param>
-    /// <param name="location"></param>
     /// <returns></returns>
-    [HttpGet("exists")]
-    [HttpGet("exists/{location}")]
+    [HttpGet]
+    [HttpGet("{locationId:int}")]
     [Produces(MediaTypeNames.Application.Json)]
-    [ProducesResponseType((int)HttpStatusCode.OK)]
-    [ProducesResponseType((int)HttpStatusCode.NoContent)]
+    [ProducesResponseType(typeof(FolderModel), (int)HttpStatusCode.OK)]
     [ProducesResponseType((int)HttpStatusCode.BadRequest)]
     [SwaggerOperation(Tags = new[] { "Storage" })]
-    public IActionResult FolderExists([FromQuery] string? path, [FromRoute] string? location = "capture")
+    public IActionResult GetFolder([FromRoute] int? locationId, [FromQuery] string? path)
     {
-        var safePath = Path.Combine(_options.GetRootPath(location), path.MakeRelativePath());
-        var exists = Directory.Exists(safePath);
-        return exists ? new OkResult() : new NoContentResult();
+        path = String.IsNullOrWhiteSpace(path) ? "" : HttpUtility.UrlDecode(path).MakeRelativePath();
+        if (locationId != null)
+        {
+            // TODO: Handle multiple storage locations.
+            var connection = _connection.GetConnection(locationId.Value);
+            if (connection?.ConnectionType == ConnectionType.SSH)
```

Comment:
I couldn't think of a quick and clean way to reduce duplication of code in each of the endpoints regrettably.  They all have a lot in common and if I had more time I'd come up with one.